src/utils/format.ts

```diff
@@ -15,27 +27,159 @@ function _getLocaleSymbols(): { thousands: string; decimals: string } {
+  }
+
+  const amountBeforePrecisionCheck = _formatNumber(integerPart.toString(10)) + DECIMALS_SYMBOL + decimalsPadded
+  return adjustPrecision(amountBeforePrecisionCheck, finalPrecision).replace(/0+$/, '')
+}
+
+function _decomposeBn(amount: BN, amountPrecision: number, decimals: number): { integerPart: BN; decimalPart: BN, decimalsPadded: string } {
   if (decimals > amountPrecision) {
     throw new Error('The decimals cannot be bigger than the precision')
   }
-  const amountRaw = amount.divRound(TEN.pow(new BN(amountPrecision - decimals)))
-  const integerPart = amountRaw.div(TEN.pow(new BN(decimals))) // 165000 / 10000 = 16
-  const decimalPart = amountRaw.mod(TEN.pow(new BN(decimals))) // 165000 % 10000 = 5000
   // Discard the decimals we don't need
   //  i.e. for WETH (precision=18, decimals=4) --> amount / 1e14
   //        1, 18:  16.5*1e18 ---> 165000
-  return { integerPart, decimalPart }
+  const amountRaw = amount.divRound(TEN.pow(new BN(amountPrecision - decimals)))
+  const integerPart = amountRaw.div(TEN.pow(new BN(decimals))) // 165000 / 10000 = 16
+  const decimalPart = amountRaw.mod(TEN.pow(new BN(decimals))) // 165000 % 10000 = 5000
+  const decimalsPadded = decimalPart.toString(10).padStart(decimals, '0')
+
+  return { integerPart, decimalPart, decimalsPadded }
+}
+
+interface SmartFormatParams<T> extends Exclude<FormatAmountParams<T>, 'thousandSeparator' | 'isLocaleAware'> {
+  smallLimit?: string
+}
+
+/**
+ * formatSmart
+ * @description prettier formatting based on Gnosis Safe - uses same signature as formatAmount
+ * @param amount
+ * @param amountPrecision
+ */
+export function formatSmart(amount: BN, amountPrecision: number): string
+export function formatSmart(amount: null | undefined, amountPrecision: number): null
+export function formatSmart(params: SmartFormatParams<BN>): string
+export function formatSmart(params: SmartFormatParams<null | undefined>): null
+export function formatSmart(
+  params: SmartFormatParams<BN | null | undefined> | BN | null | undefined,
+  _amountPrecision?: number,
+): string | null {
+  /*
+    1. integer part in Billion or Trillion becomes abbreviated w/4 decimals + decimals are DROPPED
+        ==> e.g 1.2546T
+    2. everything under is shown as is, with local thousands separator and 4 decimal points
+        ==> e.g 125,456,777.8888
+    3. anything under "smallLimit" is shown as < ${smallLimit}
+        ==> < 0.0001
+  */
+  let amount: BN
+  let precision: number
+
+  let decimals = DEFAULT_DECIMALS
+  let smallLimit = DEFAULT_SMALL_LIMIT
+
+  if (!params || ('amount' in params && !params.amount)) return null
+
+  if (BN.isBN(params)) {
+    amount = params
+    precision = _amountPrecision as number
+  } else {
+    amount = params.amount as BN
+    precision = params.precision
+    decimals = params.decimals ?? decimals
+    smallLimit = params.smallLimit ?? smallLimit
+  }
+
+  // amount is already zero
+  if (amount.isZero()) return amount.toString(10)
```

Comment:
The _Needless Optimization Group_ reviewed you code and made the following suggestion:

Let's save 1 function call!

```suggestion
  if (amount.isZero()) return '0'
```
/shrug

Reply:
\o/ great idea